src/types/st-array.ts

```diff
@@ -1,37 +1,83 @@
-import { makeClass } from "../utils/make-class";
-import { ensureArrayLikeIs, SerializedType } from "./serialized-type";
+import { SerializedTypeClass } from "./serialized-type";
 import { STObject } from "./st-object";
+import { BinaryParser } from "../serdes/binary-parser";
 
-const STArray = makeClass(
-  {
-    mixins: SerializedType,
-    inherits: Array,
-    statics: {
-      fromParser(parser) {
-        const array = new STArray();
-        while (!parser.end()) {
-          const field = parser.readField();
-          if (field.name === "ArrayEndMarker") {
-            break;
-          }
-          const outer = new STObject();
-          outer[field.name] = parser.readFieldValue(field);
-          array.push(outer);
-        }
-        return array;
-      },
-      from(value) {
-        return ensureArrayLikeIs(STArray, value).withChildren(STObject);
```

Comment:
I think the idea originally here was that there was going to be classes which would only ever be constructed if they were valid (and thus could be passed around as so), but also to support plain javascript objects. I don't think it got that far though. 

This was going to be used for more than just serializing/deserializing though. I wrote this originally for a rippled test suite written in JS, and wanted it all completely self contained, so wasn't using ripple-lib. We ended up using it for this library though, and only exposing string -> object and vice versa ...